### PR TITLE
global typescript option

### LINKED ISSUE #1114: Feature Request: Global TypeScript option in Mitosis Config
Currently you have to define `typescript: true`  inside `options` per target, but it would be nice if the config supported a global `typescript: true` flag that would apply to all relevant targets.

### PR DESCRIPTION
**Pull request by moaoa**
## Description

Add a short description of:
- what changes you made, 
- why you made them, and 
- any other context that you think might be helpful for someone to better understand what is contained in this pull request. 

closes #1114

### CODE REVIEW

**Comment by samijaber** [comment]
Please lint your PR: https://github.com/BuilderIO/mitosis/tree/main/developer#pre-submit

**Comment by moaoa** [comment]
Done 👍

File: packages/core/src/types/config.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export type GeneratorOptions = {
 };
 
 export type MitosisConfig = {
+  commonOptions: { typescript: boolean };
```
Comment (samijaber) [inline packages/core/src/types/config.ts:18]: We want all of this to be optional

```suggestion
  commonOptions?: { typescript?: boolean };
```

**Comment by ohkimur** [comment]
@moaoa @samijaber I don't think this works with tsx files for some reason? I'm I missing something?

**Comment by ohkimur** [comment]
@samijaber Can you create an issue for this particular one, please? I'll create issues when I'll find other bugs in the future.

**Pull request merged**